website/docs/r/cdn_frontdoor_custom_domain.html.markdown

```diff
@@ -0,0 +1,135 @@
+---
+subcategory: "CDN"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_cdn_frontdoor_profile_custom_domain"
+description: |-
+  Manages a Frontdoor Custom Domain.
+---
+
+# azurerm_cdn_frontdoor_profile_custom_domain
+
+Manages a Frontdoor Custom Domain.
+
+!>**IMPORTANT:** To delete a Custom Domain you must first disassociate the Custom Domain from the Frontdoor Route by removing the Custom Domains Resource ID from the Frontdoor Routes `cdn_frontdoor_custom_domain_ids` field. Once the Frontdoor Route has been updated, and the change to the Route has been provisioned to Azure, you may then remove the Custom Domain from your configuration file.
+
+!>**IMPORTANT:** If you are using Terraform to manage your DNS Auth and DNS CNAME records for your Custom Domain you will need to add configuration blocks for both the `azurerm_dns_txt_record`(see the `Example DNS Auth TXT Record Usage` below) and the `azurerm_dns_cname_record`(see the `Example CNAME Record Usage` below) to your configuration file.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-cdn-frontdoor"
+  location = "West Europe"
+}
+
+resource "azurerm_dns_zone" "example" {
+  name                = "afdx-terraform.azfdtest.xyz "
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_cdn_frontdoor_profile" "example" {
+  name                = "example-profile"
+  resource_group_name = azurerm_resource_group.example.name
+}
+
+resource "azurerm_cdn_frontdoor_custom_domain" "example" {
+  name                     = "example-customDomain"
+  cdn_frontdoor_profile_id = azurerm_cdn_frontdoor_profile.example.id
+  dns_zone_id              = azurerm_dns_zone.example.id
+  host_name                = "contoso.com"
+
+  tls {
+    certificate_type    = "ManagedCertificate"
+    minimum_tls_version = "TLS12"
+  }
+}
+```
+## Example DNS Auth TXT Record Usage
+
+```hcl
+resource "azurerm_dns_txt_record" "example" {
+  name                = join(".", ["_dnsauth", "contoso"])
+  zone_name           = azurerm_dns_zone.example.name
+  resource_group_name = azurerm_resource_group.example.name
+  ttl                 = 3600
+
+  record {
+    value = azurerm_cdn_frontdoor_custom_domain.example.validation_token
+  }
+}
+```
+
+## Example CNAME Record Usage
+
+!>**IMPORTANT:** You **must** include the `depends_on` meta-argument which references both the `azurerm_cdn_frontdoor_route` and the `azurerm_cdn_frontdoor_security_policy` that are associated with your Custom Domain.
```

Comment:
.. why? If we're missing a lock we should add that, if it's for ordering we should explain that here

Reply:
It is for ordering, you cannot write the DNS CNAME record until all of the resources are "associated" in FrontDoor which is the reason for the `depends_on` args. Fixed.